From 181f1c396e70cbaac4b013a067b8b870f1fb6642 Mon Sep 17 00:00:00 2001
From: Marcin Niestroj <m.niestroj@emb.dev>
Date: Tue, 16 Apr 2024 17:57:43 +0200
Subject: [PATCH] drivers: net: nsos: pass addrlen by value in
 sockaddr_to_nsos_mid()

There is no reason to pass addrlen by pointer, since it is a read-only in
the context of sockaddr_to_nsos_mid().

Signed-off-by: Marcin Niestroj <m.niestroj@emb.dev>
---
 drivers/net/nsos_sockets.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/net/nsos_sockets.c b/drivers/net/nsos_sockets.c
index 2e1ff4589d1..e59d940b4d2 100644
--- a/drivers/net/nsos_sockets.c
+++ b/drivers/net/nsos_sockets.c
@@ -372,7 +372,7 @@ static int nsos_ioctl(void *obj, unsigned int request, va_list args)
 	return -EINVAL;
 }
 
-static int sockaddr_to_nsos_mid(const struct sockaddr *addr, socklen_t *addrlen,
+static int sockaddr_to_nsos_mid(const struct sockaddr *addr, socklen_t addrlen,
 				struct nsos_mid_sockaddr **addr_mid, size_t *addrlen_mid)
 {
 	if (!addr || !addrlen) {
@@ -389,7 +389,7 @@ static int sockaddr_to_nsos_mid(const struct sockaddr *addr, socklen_t *addrlen,
 		struct nsos_mid_sockaddr_in *addr_in_mid =
 			(struct nsos_mid_sockaddr_in *)*addr_mid;
 
-		if (*addrlen < sizeof(*addr_in)) {
+		if (addrlen < sizeof(*addr_in)) {
 			return -NSOS_MID_EINVAL;
 		}
 
@@ -407,7 +407,7 @@ static int sockaddr_to_nsos_mid(const struct sockaddr *addr, socklen_t *addrlen,
 		struct nsos_mid_sockaddr_in6 *addr_in_mid =
 			(struct nsos_mid_sockaddr_in6 *)*addr_mid;
 
-		if (*addrlen < sizeof(*addr_in)) {
+		if (addrlen < sizeof(*addr_in)) {
 			return -NSOS_MID_EINVAL;
 		}
 
@@ -477,7 +477,7 @@ static int nsos_bind(void *obj, const struct sockaddr *addr, socklen_t addrlen)
 	size_t addrlen_mid;
 	int ret;
 
-	ret = sockaddr_to_nsos_mid(addr, &addrlen, &addr_mid, &addrlen_mid);
+	ret = sockaddr_to_nsos_mid(addr, addrlen, &addr_mid, &addrlen_mid);
 	if (ret < 0) {
 		goto return_ret;
 	}
@@ -501,7 +501,7 @@ static int nsos_connect(void *obj, const struct sockaddr *addr, socklen_t addrle
 	size_t addrlen_mid;
 	int ret;
 
-	ret = sockaddr_to_nsos_mid(addr, &addrlen, &addr_mid, &addrlen_mid);
+	ret = sockaddr_to_nsos_mid(addr, addrlen, &addr_mid, &addrlen_mid);
 	if (ret < 0) {
 		goto return_ret;
 	}
@@ -653,7 +653,7 @@ static ssize_t nsos_sendto(void *obj, const void *buf, size_t len, int flags,
 
 	flags_mid = ret;
 
-	ret = sockaddr_to_nsos_mid(addr, &addrlen, &addr_mid, &addrlen_mid);
+	ret = sockaddr_to_nsos_mid(addr, addrlen, &addr_mid, &addrlen_mid);
 	if (ret < 0) {
 		goto return_ret;
 	}
-- 
2.45.2

